plutus-core/plutus-ir/test/PlutusIR/Generators/QuickCheck/Tests.hs

```diff
@@ -184,3 +194,22 @@ prop_noTermShrinkLoops = withMaxSuccess 10 $
   forAllDoc "ty,tm" genTypeAndTerm_
     (\(ty', tm') -> filter ((/= tm') . snd) $ shrinkClosedTypedTerm (ty', tm')) $ \(ty, tm) ->
   tm `notElem` map snd (shrinkClosedTypedTerm (ty, tm))
+
+-- | Check that evaluation of the given term doesn't fail with a structural error.
```

Comment:
It took some digging through the code to discover how this is checking something, so more explanation might be helpful.  The logic of  [extractEvaluationResult](https://github.com/IntersectMBO/plutus/blob/200ad0ffe59f841b6cae82995d5f080ba71aab0d/plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/Exception.hs#L98) seems a bit obscure because it's somewhat unexpectedly using both `Left` and `Right` to represent failures and then `unsafeToEvaluationResult` is turning one of those into an actual exception using `unsafeFromEither`, and that's what's coming into play here.  It might be easier to understand if all of this was done in a more explicit way, or at least if there was an explanation here of what's going on.

Reply:
I've now made it

```haskell
-- See the Haddock of 'EvaluationError' for what structural and operational errors are.
-- See Note [Ignoring context in OperationalEvaluationError].
-- | Preserve the contents of an 'StructuralEvaluationError' as a 'Left' and turn an
-- 'OperationalEvaluationError' into a @Right EvaluationFailure@ (thus erasing the content of the
-- error in the latter case).
splitStructuralOperational
    :: Either (EvaluationException structural operational term) a
    -> Either (ErrorWithCause structural term) (EvaluationResult a)
```

and

```haskell
noStructuralErrors :: UPLC.Term Name DefaultUni DefaultFun () -> IO ()
noStructuralErrors term =
  -- Throw on a structural evaluation error and succeed on both an operational evaluation error and
  -- evaluation success.
  void . evaluate . unsafeSplitStructuralOperational . fst $ do
    <...>
```

I feel like this is sufficiently clear now. `extractEvaluationResult` was just some old name back from the days when the distinction between structural and operational errors wasn't really defined.